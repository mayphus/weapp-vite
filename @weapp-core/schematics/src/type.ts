import type { App as _App, Component as _Component, Page as _Page, Sitemap as _Sitemap, Theme as _Theme } from './type.auto'

export type App = _App
export type Component = _Component
export type Page = _Page
export type Sitemap = _Sitemap
export type Theme = _Theme
