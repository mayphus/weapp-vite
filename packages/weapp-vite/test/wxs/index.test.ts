// import path from 'pathe'
import type { ProjectOptions } from 'ts-morph'
// import { defu } from 'defu'
import { ModuleKind, Project, ScriptTarget } from 'ts-morph'

const sharedProjectOptions: ProjectOptions = {
  compilerOptions: {
    target: ScriptTarget.ES5,
    removeComments: true,
    esModuleInterop: false,
    allowSyntheticDefaultImports: false,
    // module: 'CommonJS',
    module: ModuleKind.CommonJS,
    // alwaysStrict: false,
    // strict: false,

  },
  useInMemoryFileSystem: true,
}

// https://github.com/microsoft/TypeScript/issues/30166
// https://github.com/microsoft/TypeScript/pull/32083
describe('wxs', () => {
  it('should ', async () => {
    const project = new Project(sharedProjectOptions)

    // const sourceFile = project.addSourceFileAtPathIfExists(path.resolve(__dirname, '../fixtures/wxs/case0.wxs.ts'))
    //     const sourceFile = project.createSourceFile('file.ts', `export const foo = "'hello world' from comm.wxs";
    // export const bar = function (d: string) {
    //   return d;
    // }
    // `, { overwrite: true })
    const sourceFile = project.createSourceFile('file.ts', `export const foo = "'hello world' from comm.wxs";
export const bar = function (d: string) {
  return d;
}
`)
    // project.getSourceFiles().forEach((sourceFile) => {
    //   const exportsModule = sourceFile.getVariableDeclaration('__esModule')
    //   if (exportsModule) {
    //     exportsModule.remove()
    //   }
    //   sourceFile.saveSync()
    // })
    // await sourceFile?.save()
    // const res = await sourceFile.emit()
    const exportsModule = sourceFile.getVariableDeclaration('__esModule')
    if (exportsModule) {
      exportsModule.remove()
    }
    await sourceFile.save()
    const emitOutput = sourceFile.getEmitOutput()
    // emitOutput.getEmitSkipped() // returns: boolean
    for (const outputFile of emitOutput.getOutputFiles()) {
      // console.log(outputFile.getFilePath(), outputFile.getWriteByteOrderMark(), outputFile.getText())
      expect(outputFile.getFilePath()).toMatchSnapshot('FilePath')
      expect(outputFile.getText()).toMatchSnapshot('Text')
    }
    // console.log(res.compilerObject.emittedFiles)
    // console.log(res)
    // await project.emit()
    // const fs = project.getFileSystem()
    //  expect(await fs.readFile('file.ts')).toBe('console.log(5);')
    // const res = await project.emit()
    // console.log(res)
  })

  it('should 0', () => {
    const project = new Project(sharedProjectOptions)
    const sourceFile = project.createSourceFile('file.ts', `var foo = "'hello world' from comm.wxs";
var bar = function(d) {
  return d;
}
module.exports = {
  foo: foo,
  bar: bar
};
`)

    const emitOutput = sourceFile.getEmitOutput()
    for (const outputFile of emitOutput.getOutputFiles()) {
      expect(outputFile.getFilePath()).toMatchSnapshot('FilePath')
      expect(outputFile.getText()).toMatchSnapshot('Text')
    }
  })

  it('should 1', () => {
    const project = new Project(sharedProjectOptions)
    const sourceFile = project.createSourceFile('file.ts', `var tools = require("./tools.wxs");

console.log(tools.FOO);
console.log(tools.bar("logic.wxs"));
console.log(tools.msg);
`)

    const emitOutput = sourceFile.getEmitOutput()
    for (const outputFile of emitOutput.getOutputFiles()) {
      expect(outputFile.getFilePath()).toMatchSnapshot('FilePath')
      expect(outputFile.getText()).toMatchSnapshot('Text')
    }
  })

  it('should 2', () => {
    const project = new Project(sharedProjectOptions)
    const sourceFile = project.createSourceFile('file.ts', `var some_msg = "hello world";
module.exports = {
  msg : some_msg,
}
`)

    const emitOutput = sourceFile.getEmitOutput()
    for (const outputFile of emitOutput.getOutputFiles()) {
      expect(outputFile.getFilePath()).toMatchSnapshot('FilePath')
      expect(outputFile.getText()).toMatchSnapshot('Text')
    }
  })

  it('should 3', () => {
    const project = new Project(sharedProjectOptions)
    const sourceFile = project.createSourceFile('file.ts', `// 方法一：单行注释

/*
方法二：多行注释
*/

/*
方法三：结尾注释。即从 /* 开始往后的所有 WXS 代码均被注释

var a = 1;
var b = 2;
var c = "fake";
`)

    const emitOutput = sourceFile.getEmitOutput()
    for (const outputFile of emitOutput.getOutputFiles()) {
      expect(outputFile.getFilePath()).toMatchSnapshot('FilePath')
      expect(outputFile.getText()).toMatchSnapshot('Text')
    }
  })

  it('should 4', () => {
    const project = new Project(sharedProjectOptions)
    const sourceFile = project.createSourceFile('file.ts', `const foo = "'hello world' from tools.wxs";
const bar = function (d) {
  return d;
}
module.exports = {
  FOO: foo,
  bar: bar,
};
module.exports.msg = "some msg";
`)

    const emitOutput = sourceFile.getEmitOutput()
    for (const outputFile of emitOutput.getOutputFiles()) {
      expect(outputFile.getFilePath()).toMatchSnapshot('FilePath')
      expect(outputFile.getText()).toMatchSnapshot('Text')
    }
  })

  it('should 5', () => {
    const project = new Project(sharedProjectOptions)
    const sourceFile = project.createSourceFile('file.ts', `export const foo = "'hello world' from tools.wxs";
export const bar = function (d) {
  return d;
}

export const msg = "some msg";
`)

    const emitOutput = sourceFile.getEmitOutput()
    for (const outputFile of emitOutput.getOutputFiles()) {
      expect(outputFile.getFilePath()).toMatchSnapshot('FilePath')
      expect(outputFile.getText()).toMatchSnapshot('Text')
    }
  })
})
