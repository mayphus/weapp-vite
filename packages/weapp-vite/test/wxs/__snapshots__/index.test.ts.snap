// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`wxs > should  > FilePath 1`] = `"/file.js"`;

exports[`wxs > should  > Text 1`] = `
""use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.bar = exports.foo = void 0;
exports.foo = "'hello world' from comm.wxs";
var bar = function (d) {
    return d;
};
exports.bar = bar;
"
`;

exports[`wxs > should 0 > FilePath 1`] = `"/file.js"`;

exports[`wxs > should 0 > Text 1`] = `
"var foo = "'hello world' from comm.wxs";
var bar = function (d) {
    return d;
};
module.exports = {
    foo: foo,
    bar: bar
};
"
`;

exports[`wxs > should 1 > FilePath 1`] = `"/file.js"`;

exports[`wxs > should 1 > Text 1`] = `
"var tools = require("./tools.wxs");
console.log(tools.FOO);
console.log(tools.bar("logic.wxs"));
console.log(tools.msg);
"
`;

exports[`wxs > should 2 > FilePath 1`] = `"/file.js"`;

exports[`wxs > should 2 > Text 1`] = `
"var some_msg = "hello world";
module.exports = {
    msg: some_msg,
};
"
`;

exports[`wxs > should 3 > FilePath 1`] = `"/file.js"`;

exports[`wxs > should 3 > Text 1`] = `""`;

exports[`wxs > should 4 > FilePath 1`] = `"/file.js"`;

exports[`wxs > should 4 > Text 1`] = `
"var foo = "'hello world' from tools.wxs";
var bar = function (d) {
    return d;
};
module.exports = {
    FOO: foo,
    bar: bar,
};
module.exports.msg = "some msg";
"
`;

exports[`wxs > should 5 > FilePath 1`] = `"/file.js"`;

exports[`wxs > should 5 > Text 1`] = `
""use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.msg = exports.bar = exports.foo = void 0;
exports.foo = "'hello world' from tools.wxs";
var bar = function (d) {
    return d;
};
exports.bar = bar;
exports.msg = "some msg";
"
`;
